asdf/commands/tests/test_edit.py

```diff
@@ -0,0 +1,231 @@
+   accomodate future edits.  If a block index existed in the original ASDF
+   file, it will need to be recomputed and if one didn't exist, it will be
+   added to the resultant ASDF file.  
+"""
+
+
+
+def create_base_asdf(tmpdir, version):
+    """
+    In the test temp directory, create a base ASDF file to edit
+    and test against.
+    """
+    seq = np.arange(100)
+
+    # Store the data in an arbitrarily nested dictionary
+    tree = {
+        "foo": 42,
+        "name": "Monty",
+        "sequence": seq,
+    }
+
+    fname = "test_edit_base.asdf"
+    oname = os.path.join(tmpdir, fname)
+    if os.path.exists(oname):
+        os.remove(oname)
+    af = asdf.AsdfFile(tree,version=version)
+    af.write_to(oname)
+
+    return oname
+
+
+def create_edit_equal(base_yaml):
+    """
+    The YAML from the base ASDF file will have a 'foo' value of 42.  Create
+    an edited YAML file with this value being 41.  This will create an edited
+    YAML file with the same number of characters in the YAML section as was in
+    the original ASDF file.
+    """
+    with open(base_yaml, "r") as fd:
+        lines = fd.readlines()
+
+    base, ext = os.path.splitext(base_yaml)
+    oname = f"{base}_edit_equal.yaml"
+    if os.path.exists(oname):
+        os.remove(oname)
+    with open(oname, "w") as fd:
+        for l in lines:
+            if "foo" in l:
+                print("foo: 41", file=fd)  #  Change a value
+            else:
+                fd.write(l)
+
+    return oname
+
+
+def create_edit_smaller(base_yaml):
+    """
+    The YAML from the base ASDF file will have a 'foo' value of 42.  Create
+    an edited YAML file with this value being 41.  This will create an edited
+    YAML file with the same number of characters in the YAML section as was in
+    the original ASDF file.
+    """
+    with open(base_yaml, "r") as fd:
+        lines = fd.readlines()
+
+    base, ext = os.path.splitext(base_yaml)
+    oname = f"{base}_edit_smaller.yaml"
+    if os.path.exists(oname):
+        os.remove(oname)
+    with open(oname, "w") as fd:
+        for l in lines:
+            if "foo" in l:
+                print("foo: 2", file=fd)  #  Change a value
+            else:
+                fd.write(l)
+
+    return oname
+
+
+def create_edit_larger(base_yaml):
+    """
+    The YAML from the base ASDF file will have a 'foo' value.  After this
+    line, add another line.  This will create an edited YAML file that will
+    have more characters than the YAML portion of the original ASDF file.
+    """
+    with open(base_yaml, "r") as fd:
+        lines = fd.readlines()
+
+    base, ext = os.path.splitext(base_yaml)
+    oname = f"{base}_edit_larger.yaml"
+    if os.path.exists(oname):
+        os.remove(oname)
+    with open(oname, "w") as fd:
+        for l in lines:
+            fd.write(l)
+            if "foo" in l:
+                print("bar: 13", file=fd)  # Add a line
+
+    return oname
+
+
+def copy_base_asdf_equal(base_asdf):
```

Comment:
Another good opportunity for code reuse here, we can just pass `oname` as a parameter to the function.

Reply:
This has been refactored out of the test.